servicereportpkg/utils.py

```diff
@@ -343,7 +343,7 @@ def restart_service(service):
 
 
 def update_grub():
-    command = ["grub2-mkconfig", "-o", "/boot/grub2/grub.cfg"]
+    command = ["grub2-mkconfig", "-o", "/boot/grub2/grub.cfg", "--update-bls-cmdline"]
```

Comment:
Is this option available with different distribution releases? If not it will be better to make this conditional.

Reply:
@sacsant That's a good point .  This option is not available on SLES .  Better to add conditional check for RHEL93 and above to support this option . And , worth exploring other distros as well

Reply:
@sacsant  @sourabhjains  Please review  the changes . As per the observations with other Linux distributions , we need  to add the  option "--update-bls-cmdline" for only RHEL  ( 9.3  and above versions)

Reply:
Instead of depending on version/release, do you think it will be easier to use following logic?
if grub tool provides --update-bls-cmdline
     invoke grub2 with this option
else
     use existing grub2 command logic

This should support all future distros as well without adding explicit check for version/releases.

Reply:
@SACHIN-BAPPALIGE I liked the @sacsant suggestion of finding the support for --update-bls-cmdline and using it.

Reply:
@sacsant @sourabhjains Thanks for the suggestion . Then this command can be used to check whether or not  grub tool provides --update-bls-cmdline .

```
# grub2-mkconfig  -h | grep "update-bls-cmdline"
  --update-bls-cmdline    overwrite BLS cmdline args with default args
#
```

Reply:
Not an elegant way to find about the feature but lets go with this approach for now.

Reply:
@SACHIN-BAPPALIGE I am thinking of using grubby for RHEL to update the crashkernel value. So are you fine with me taking this patch forward and making you co-developer?

Reply:
@sourabhjains Please go ahead with above suggested method

Reply:
Thanks.